libmozevent/mercurial.py

```diff
@@ -454,7 +453,10 @@ async def handle_build(self, repository, build):
             if isinstance(error_log, bytes):
                 error_log = error_log.decode("utf-8")
 
-            if "push failed on remote" in error_log.lower():
+            if (
+                "push failed on remote" in error_log.lower()
+                or "stream ended unexpectedly" in error_log.lower()
+            ):
```

Comment:
Could we also add "EOF occurred in violation of protocol", which happened in https://phabricator.services.mozilla.com/B527171.